pandas/core/indexing.py

```diff
@@ -1426,14 +1427,231 @@ class _LocIndexer(_LocationIndexer):
     - A list or array of labels, e.g. ``['a', 'b', 'c']``.
     - A slice object with labels, e.g. ``'a':'f'`` (note that contrary
       to usual python slices, **both** the start and the stop are included!).
```

Comment:
May be we could use the `.. warning::` directive for this comment (instead of a note in brackets ended with the exclamation mark).